FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download_Multithreaded/DemoTasks/S3DownloadMultithreadedHTTPExample.c

```diff
@@ -0,0 +1,1142 @@
+ * @param[in] xStart The position of the first byte in the range.
+ * @param[in] xEnd The position of the last byte in the range, inclusive.
+ *
+ * @return pdPASS if request successfully enqueued; pdFAIL otherwise.
+ */
+static BaseType_t prvRequestS3ObjectRange( const HTTPRequestInfo_t * pxRequestInfo,
+                                           const size_t xStart,
+                                           const size_t xEnd );
+
+/**
+ * @brief Check for a task notification.
+ *
+ * @param[in] pulNotification Pointer holding the notification value.
+ * @param[in] ulExpectedBits Bits to wait for.
+ * @param[in] xClearBits If bits should be cleared.
+ *
+ * @return pdTRUE if notification received; pdFALSE otherwise.
+ */
+static BaseType_t prvCheckNotification( uint32_t * pulNotification,
+                                        uint32_t ulExpectedBits,
+                                        BaseType_t xClearBits );
+
+/**
+ * @brief Enqueue a request for the size of the S3 object specified in pcPath.
+ *
+ * @param[in] pxRequestInfo The #HTTPRequestInfo_t for configuring the request.
+ *
+ * @return pdFAIL on failure; pdPASS on success.
+ */
+static BaseType_t prvGetS3ObjectFileSize( const HTTPRequestInfo_t * pxRequestInfo );
+
+/**
+ * @brief Enqueue HTTP range requests onto the request queue continuously, until
+ * the entire length of the file has been requested.
+ *
+ * @param[in] pvArgs Parameters as passed at the time of task creation. Not used
+ * in this example.
+ */
+static void prvRequestTask( void * pvArgs );
+
+/**
+ * @brief Read and interpret the server response corresponding to the file size
+ * request. Called by the response task.
+ *
+ * @return pdFAIL on failure; pdPASS on success.
+ */
+static BaseType_t prvReadFileSize( void );
+
+/**
+ * @brief Log and interpret responses present on the response queue
+ * continuously, until the length of the file is downloaded.
+ *
+ * @param[in] pvArgs Parameters as passed at the time of task creation. Not used
+ * in this example.
+ */
+static void prvResponseTask( void * pvArgs );
+
+/**
+ * @brief Services HTTP requests from the request queue and writes corresponding
+ * responses to the response queue. Called by the main task.
+ *
+ * @return pdFAIL on failure; pdPASS on success.
+ */
+static BaseType_t prvDownloadLoop( void );
+
+/*-----------------------------------------------------------*/
+
+/*
+ * @brief Create task to demonstrate the HTTP API over a server-authenticated
+ * network connection with a server.
+ */
+void vStartSimpleHTTPDemo( void )
+{
+    /* This example uses one application task to process the request queue for
+     * HTTP operations, and creates additional tasks to add operations to that
+     * queue and interpret server responses. */
+    xTaskCreate( prvHTTPDemoTask,          /* Function that implements the task. */
+                 "DemoTask",               /* Text name for the task - only used for debugging. */
```

Comment:
It will be better to keep a consistent naming. If we are calling it main task in comments, it is good to name it the same here.